metastore/src/main/scala/com/gu/tableversions/metastore/VersionPaths.scala

```diff
@@ -31,8 +32,17 @@ object VersionPaths {
       partitionPath
     else {
       def normalised(path: String): String = if (path.endsWith("/")) path else path + "/"
+
       def versioned(path: String): String = s"$path${version.label}"
+
       new URI(versioned(normalised(partitionPath.toString)))
     }
 
+  def flattenMap(partitions: List[Partition]): Map[String, String] = {
```

Comment:
I think this produces Hadoop configuration entries. Maybe a more descriptive name and/or a comment would be useful here?

Reply:
See comment on the filesystem proxy, but I would expect the version for each partition to go in here as well.

Reply:
This method (name quite possibly unintuitive aside) is turning `Partition`s into strings like `part1=value1/part2=value2`. It does not per se add it to spark's configuration itself (`VersionedDataset.writeVersionedPartitions()` does that)

I'll still give it a better name for now.

Reply:
I know it doesn't _add_ to the configuration, but it is producing properties for the hadoop config, no? So it seems logically to belong with the details of the `VersionedFilesystem` code.

Reply:
Actually, I am sort of wrong. It does have a `KEY_PREFIX` and sits already half way through to preparing stuf for spark's config

Reply:
I agree, the name of this def is probably not descriptive enough.

Reply:
Renamed and moved to FS's companion object